discretize/base/base_mesh.py

```diff
@@ -748,798 +672,2926 @@ def project_edge_vector(self, edge_vectors):
+        # >>> ax2.set_yticks([])
+        # >>> ax2.set_ylabel("")
+        # >>> ax2.set_title("Laplacian at nodes", fontsize=14)
+        # >>> plt.show()
+
+        # The nodal Laplacian operator is a sparse matrix that maps
+        # from nodes to nodes. To demonstrate this, we construct
+        # a small 2D mesh. We then show the ordering of the nodes
+        # and a spy plot illustrating the sparsity of the operator.
+
+        # >>> mesh = TensorMesh([[(1, 4)], [(1, 4)]])
+        # >>> fig = plt.figure(figsize=(12, 6))
+        # >>> ax1 = fig.add_subplot(211)
+        # >>> mesh.plot_grid(ax=ax1)
+        # >>> ax1.set_title("Ordering of the Nodes", fontsize=14, pad=15)
+        # >>> ax1.plot(mesh.nodes[:, 0], mesh.nodes[:, 1], "ro", markersize=8)
+        # >>> for ii, loc in zip(range(mesh.nN), mesh.nodes):
+        # ...     ax1.text(loc[0] + 0.05, loc[1] + 0.02, "{0:d}".format(ii), color="r")
+        # >>> ax1.set_xticks([])
+        # >>> ax1.set_yticks([])
+        # >>> ax1.spines['bottom'].set_color('white')
+        # >>> ax1.spines['top'].set_color('white')
+        # >>> ax1.spines['left'].set_color('white')
+        # >>> ax1.spines['right'].set_color('white')
+        # >>> ax1.set_xlabel('X', fontsize=16, labelpad=-5)
+        # >>> ax1.set_ylabel('Y', fontsize=16, labelpad=-15)
+        # >>> ax1.legend(
+        # ...     ['Mesh', '$\\mathbf{\\phi}$ (nodes)'],
+        # ...     loc='upper right', fontsize=14
+        # ... )
+        # >>> ax2 = fig.add_subplot(212)
+        # >>> ax2.spy(mesh.nodal_laplacian)
+        # >>> ax2.set_title("Spy Plot", fontsize=14, pad=5)
+        # >>> ax2.set_ylabel("Node Index", fontsize=12)
+        # >>> ax2.set_xlabel("Node Index", fontsize=12)
+        raise NotImplementedError(f"nodal_laplacian not implemented for {type(self)}")
 
-        This property returns a tuple containing the number of x-faces
-        along each axis direction.
+    @property
+    def stencil_cell_gradient(self):
+        """Stencil for cell gradient operator (cell centers to faces)
+
+        This property constructs a differencing operator that acts on
+        cell centered quantities. The operator takes the difference between
+        the values at adjacent cell centers along each axis direction,
+        and places the result on the shared face; e.g. differences
+        along the x-axis are mapped to x-faces. The operator is a sparse
+        matrix :math:`\\mathbf{G}` that can be applied as a matrix-vector
+        product to a cell centered quantity :math:`\\boldsymbol{\\phi}`, i.e.::
+
+            diff_phi = G @ phi
+
+        By default, the operator assumes zero-Neumann boundary conditions
+        on the scalar quantity. Before calling **stencil_cell_gradient** however,
+        the user can set a mix of zero Dirichlet and zero Neumann boundary
+        conditions using :py:attr:`~discretize.operators.DiffOperators.set_cell_gradient_BC`.
```

Comment:
Do changes to the boundary conditions always have to be called beforehand? or if the stencil_cell_gradient is created and then the boundary conditions are updated it would be deleted and re-created?